### DISCUSSION TITLE
Building for a translation

### DISCUSSION THREAD

**Discussion by moreirayokoyama** [google/comprehensive-rust#446] [Q&A] [answered]
Hi.
I'm trying to improve the `Brazilian Portuguese` translation for the course, but when I run the project locally I'm getting a 404 error when I try to see any translated version (for any one of the alternative translations, including `Korean`).
Is there some kind of additional dependency I have to install to make `mdbook` to support the translations? Or am I doing something wrong? (I tried to check mdbook's repo to see if by any chance there was some fix being discussed, but the feature being discussed there seems to be completely different from the one used here).
Any advice, please?

**Comment by jooyunghan** [comment, marked as answer]
Here's how I did:

```
mdbook build
MDBOOK_BOOK__LANGUAGE=ko MDBOOK_PREPROCESSOR__GETTEXT__PO_FILE=po/ko.po mdbook build -d book/ko
MDBOOK_BOOK__LANGUAGE=pt-BR MDBOOK_PREPROCESSOR__GETTEXT__PO_FILE=po/pt-BR.po mdbook build -d book/pt-BR
xdg-open book/index.html
```

Note that `mdbook build` should come first, otherwise it will blow up `ko` and `pt-BR` in it.

**Reply by moreirayokoyama** [reply]
Got it.
It really seems to create the directories and exporting files, but they're not translated. So I checked out the output for the build command, and noticed the following message:
```
MDBOOK_BOOK__LANGUAGE=pt-BR MDBOOK_PREPROCESSOR__GETTEXT__PO_FILE=po/pt-BR.po mdbook build -d book/pt-BR
2023-02-22 10:23:10 [INFO] (mdbook::book): Book building has started
2023-02-22 10:23:10 [ERROR] (mdbook::utils): Error: Unable to start the "gettext" preprocessor. Is it installed?
2023-02-22 10:23:10 [ERROR] (mdbook::utils):    Caused By: No such file or directory (os error 2)
```
Not sure of what could be the cause for this. I tried to dig into the mdBook repository to understand this gettext preprocessor and how to get it installed. I'm not sure if it is some beta feature (I had the main branch installed, but still getting the same message). I'll try to dig more into that to understand how mdBook works and have a solution. Thanks anyway.

**Reply by jooyunghan** [reply]
See https://github.com/google/comprehensive-rust/blob/main/TRANSLATIONS.md

```
$ cargo install --path i18n-helpers
```

installs `gettext` preprocessor, which translates the book.

**Reply by moreirayokoyama** [reply]
Ok. Thank you. It does work now.

**Reply by mgeisler** [reply]
Hi @moreirayokoyama, let me know if I can update the instructions in [`TRANSLATIONS.md`](https://github.com/google/comprehensive-rust/blob/main/TRANSLATIONS.md). One thing which should help is #423 since that would give you a tool with a few commands to run instead of the big command lines we have right now.
